src/Data/Fin/Properties.agda

```diff
@@ -675,6 +675,23 @@ punchOut-punchIn (suc i) {suc j} = cong suc (begin
   j                                                           ∎)
   where open ≡-Reasoning
 
+
+------------------------------------------------------------------------
+-- pinch
+------------------------------------------------------------------------
+
+pinch-surjective : ∀ {m} i (j : Fin m) → Σ[ k ∈ Fin (suc m) ] (pinch i k ≡ j)
+pinch-surjective 0F 0F = zero , refl
+pinch-surjective 0F (suc j) = (suc (suc j)) , refl
+pinch-surjective (suc i) 0F = zero , refl
+pinch-surjective (suc i) (suc j) = map suc (cong suc) (pinch-surjective i j)
+
+pinch-mono-≤ : ∀ {m} (i : Fin m) → (pinch i) Preserves _≤_ ⟶ _≤_
+pinch-mono-≤ 0F {0F} {k} 0≤n = z≤n
+pinch-mono-≤ 0F {suc j} {suc k} (s≤s j≤k) = j≤k
+pinch-mono-≤ (suc i) {0F} {k} 0≤n = z≤n
+pinch-mono-≤ (suc i) {suc j} {suc k} (s≤s j≤k) = s≤s (pinch-mono-≤ i j≤k)
```

Comment:
I don't think you need to pattern match on the `j` and `k` implicits in this proof?

Reply:
Surprisingly, you do! If you don't, agda gets very confused and yells about not being able to solve some unification problems.

Reply:
Ah yes because you're matching on `Fin (suc n)` rather than `Fin n`!